examples/3_grouped_simple/main.go

```diff
@@ -18,9 +19,9 @@ import (
 type checkMemoryAndCPU struct {
 }
 
-func (c checkMemoryAndCPU) Run(ctx gornir.Context, wg *sync.WaitGroup, jobResult chan *gornir.JobResult) {
+func (c *checkMemoryAndCPU) Run(ctx context.Context, wg *sync.WaitGroup, taskParameters *gornir.TaskParameters, jobResult chan *gornir.JobResult) {
 	// We instantiate a new object
-	result := gornir.NewJobResult(ctx)
+	result := gornir.NewJobResult(ctx, taskParameters)
```

Comment:
I'd make the name of the arguments shorter. `tp *gornir.TaskParameters` instead of `taskParameters *gornir.TaskParameters`, etc. The scope of the function is limited so you see immediately what `tp` is for example.

```go
func (c *checkMemoryAndCPU) Run(ctx context.Context, wg *sync.WaitGroup, tp *gornir.TaskParameters, jr chan *gornir.JobResult) {
	// We instantiate a new object
	result := gornir.NewJobResult(ctx)
	result := gornir.NewJobResult(ctx, tp)
```